internal/ingest/ingest.go

```diff
@@ -190,7 +190,7 @@ func (ing *Ingester) signalMetricsUpdate() {
 // metricsUpdate periodically updates metrics.  This goroutine exits when the
 // sigUpdate channel is closed, when Close is called.
 func (ing *Ingester) metricsUpdater() {
-	var hasUpdate bool
+	hasUpdate := true
```

Comment:
shouldn't this default to false?

Reply:
I purposely set it to true so that it exposes this metric (store size) on start. Note that:
1. Store size is a gauge, so we need to track what the current value is.
2. We don't expose a metric in `/metrics` until we first record it.

The correct behavior should be that the store size is known at start time, and updated periodically. Right now the latter is true, but not the former.

As it is now, we won't know the store size until we ingest something.

Does that make sense? If so I can add a comment explaining the gist of this.

Reply:
makes sense. i had confused this with a call to update metrics that was short lived.